### PR TITLE
fix: support Ctrl-Z job-control suspend

### LINKED ISSUE #252: ctrl-z to suspend
Is there a reason to prevent suspending it with ctrl-z? I was surprised it didn't work.

### PR DESCRIPTION
**Pull request by benvinegar**
## Summary

- restore Ctrl-Z job-control suspend behavior for Hunk's OpenTUI app
- follow the common TUI pattern used by apps like opencode: suspend the renderer, send `SIGTSTP` to the foreground process group, and resume on `SIGCONT`
- keep suspend support disabled on Windows where POSIX job control is unavailable
- add unit coverage for the Ctrl-Z suspend handler and edge cases

Closes #252.

## Verification

- `bun test src/core/jobControl.test.ts`
- `bun run typecheck`
- `bun run lint`
- `bun run format:check`
- manual PTY check: Ctrl-Z stops Hunk, `fg` resumes it, `q` exits cleanly
- manual PTY check: with `stty susp ^Y`, Ctrl-Z still stops Hunk, matching opencode-style app keybind behavior

*This PR description was generated by Pi using GPT-5*

### CODE REVIEW

File: src/core/jobControl.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    const renderer = getRenderer();
+    if (!renderer || renderer.isDestroyed) {
+      return false;
+    }
+
+    const resumeOnContinue = () => {
+      renderer.resume();
+    };
```
[the remark this answers is not in the document]
Comment (benvinegar) [reply src/core/jobControl.ts]: Addressed in the refactored job-control path. `SIGCONT` now returns early if the renderer has been destroyed, and I added a regression test covering a renderer destroyed between `SIGTSTP` and `SIGCONT`.

*This comment was generated by Pi using GPT-5*
